sign/ed25519/ed25519.go

```diff
@@ -131,24 +180,76 @@ func (kp *KeyPair) SignPure(message []byte) ([]byte, error) {
 	var signature [SignatureSize]byte
 	copy(signature[:paramB], R[:])
 	copy(signature[paramB:], S[:])
+
 	return signature[:], err
 }
 
-// Verify returns true if the signature is valid. Failure cases are invalid
-// signature, or when the public key cannot be decoded.
-func Verify(public PublicKey, message, signature []byte) bool {
-	if len(public) != PublicKeySize ||
-		len(signature) != SignatureSize ||
-		!isLessThanOrder(signature[paramB:]) {
-		return false
+// SignPure creates a signature of a message given a keypair.
+// This function handles unhashed messages.
+// The opts.HashFunc() must return zero to indicate the message hasn't been
+// hashed. This can be achieved by passing crypto.Hash(0) as the value for opts.
+func (kp *KeyPair) SignPure(message []byte, opts crypto.SignerOpts) ([]byte, error) {
```

Comment:
The API for SignPure, SignPh, SignWithCtx (and its verfication methods) do not need options. Because every one asks only for the parameters it requires.